src/test/kotlin/pt/up/fe/ni/website/backend/controller/EventControllerTest.kt

```diff
@@ -66,9 +81,103 @@ internal class EventControllerTest @Autowired constructor(
         }
     }
 
+    @Nested
+    @DisplayName("GET /events/{id}")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class GetEvent {
+        @BeforeAll
+        fun addEvent() {
+            repository.save(testEvent)
+        }
+
+        @Test
+        fun `should return the event`() {
+            mockMvc.get("/events/${testEvent.id}")
+                .andExpect {
+                    status { isOk() }
+                    content { contentType(MediaType.APPLICATION_JSON) }
+                    jsonPath("$.title") { value(testEvent.title) }
+                    jsonPath("$.description") { value(testEvent.description) }
+                    jsonPath("$.startDate") { value(testEvent.startDate.toJson()) }
+                    jsonPath("$.endDate") { value(testEvent.endDate.toJson()) }
+                    jsonPath("$.url") { value(testEvent.url) }
+                    jsonPath("$.location") { value(testEvent.location) }
+                    jsonPath("$.category") { value(testEvent.category) }
+                    jsonPath("$.thumbnailPath") { value(testEvent.thumbnailPath) }
+                }
+        }
+
+        @Test
+        fun `should fail if the event does not exist`() {
+            mockMvc.get("/events/1234").andExpect {
+                status { isNotFound() }
+                content { contentType(MediaType.APPLICATION_JSON) }
+                jsonPath("$.errors.length()") { value(1) }
+                jsonPath("$.errors[0].message") { value("event not found with id 1234") }
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("GET events/category/{category}")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class GetEventsByCategory {
+        private val testEvents = listOf(
+            testEvent,
+            Event(
+                "Bad event",
+                "This event was a failure",
+                TestUtils.createDate(2021, Calendar.OCTOBER, 27),
+                null,
+                null,
+                null,
+                null,
+                "https://example.com/exampleThumbnail2"
+            ),
+            Event(
+                "Mid event",
+                "This event was ok",
+                TestUtils.createDate(2022, Calendar.JANUARY, 15),
+                null,
+                null,
+                null,
+                "Other category",
+                "https://example.com/exampleThumbnail2"
+            )
+        )
+
+        @BeforeAll
+        fun addEvents() {
+            for (event in testEvents) repository.save(event)
+        }
+
+        @Test
+        fun `should return all events of the category`() {
+            mockMvc.get("/events/category/${testEvent.category}")
+                .andExpect {
+                    status { isOk() }
+                    content { contentType(MediaType.APPLICATION_JSON) }
+                    jsonPath("$.length()") { value(1) }
```

Comment:
This is already well but I'll just add a minor suggestion. Maybe it'd be better to make a test that returns 2 events, so that we're sure the system is really returning all events of that category (and not just one)

Reply:
That makes sense, I've added another event with that category.